tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
+        assert len(res[(1,)]['queue']) == 1
+        assert isinstance(res[(1,)]['queue'][0], qml.RX)
+        assert isinstance(res[(1,)]['observable'][0], qml.PauliY)
+
+        # third parameter subcircuit
+        assert len(res[(2,)]['queue']) == 3
+        assert isinstance(res[(2,)]['queue'][0], qml.RX)
+        assert isinstance(res[(2,)]['queue'][1], qml.RY)
+        assert isinstance(res[(2,)]['queue'][2], qml.CNOT)
+        assert isinstance(res[(2,)]['observable'][0], qml.Hermitian)
+        assert np.all(res[(2,)]['observable'][0].params[0] == qml.PhaseShift.generator[0])
+
+    def test_construct_subcircuit_layers(self):
+        """Test correct subcircuits constructed
+        when a layer structure exists"""
+        dev = qml.device('default.qubit', wires=3)
+
+        def circuit(params):
+            # section 1
+            qml.RX(params[0], wires=0)
+            # section 2
+            qml.RY(params[1], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            # section 3
+            qml.RX(params[2], wires=0)
+            qml.RY(params[3], wires=1)
+            qml.RZ(params[4], wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            # section 4
+            qml.RX(params[5], wires=0)
+            qml.RY(params[6], wires=1)
+            qml.RZ(params[7], wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            return qml.expval(qml.PauliX(0))
+
+        circuit = qml.QNode(circuit, dev)
+
+        params = np.ones([8])
+        circuit.construct_metric_tensor([params])
+        res = circuit._metric_tensor_subcircuits
+
+        # this circuit should split into 4 independent
+        # sections or layers when constructing subcircuits
+        assert len(res) == 4
+
+        # first layer subcircuit
+        assert len(res[(0,)]['queue']) == 0
+        assert isinstance(res[(0,)]['observable'][0], qml.PauliX)
+
+        # second layer subcircuit
+        assert len(res[(1,)]['queue']) == 1
+        assert isinstance(res[(1,)]['queue'][0], qml.RX)
+        assert isinstance(res[(1,)]['observable'][0], qml.PauliY)
+
+        # third layer subcircuit
+        layer = res[(2, 3, 4)]
+        assert len(layer['queue']) == 4
+        assert len(layer['observable']) == 3
+        assert isinstance(layer['queue'][0], qml.RX)
+        assert isinstance(layer['queue'][1], qml.RY)
+        assert isinstance(layer['queue'][2], qml.CNOT)
+        assert isinstance(layer['queue'][3], qml.CNOT)
+        assert isinstance(layer['observable'][0], qml.PauliX)
+        assert isinstance(layer['observable'][1], qml.PauliY)
+        assert isinstance(layer['observable'][2], qml.PauliZ)
+
+        # fourth layer subcircuit
+        layer = res[(5, 6, 7)]
+        assert len(layer['queue']) == 9
+        assert len(layer['observable']) == 3
+        assert isinstance(layer['observable'][0], qml.PauliX)
```

Comment:
Might as well put the full `queue` in here :laughing:

Reply:
I was serious :laughing:

Reply:
Oh oops, lol. Will do